numba/tests/test_serialize.py

```diff
@@ -205,5 +208,99 @@ def test_numba_unpickle(self):
         self.assertIs(got1, got2)
 
 
+
+class TestCloudPickleIssues(TestCase):
```

Comment:
Maybe this ought to also include the original reproducer as a test? https://github.com/numba/numba/issues/7356#issue-983961914

Reply:
Good point. Test added in https://github.com/numba/numba/pull/7388/commits/8f6ffdcafbfdbb5317605950c4f57f43dae83111